### ISSUE TITLE
Sleep Timer Feature Request

### ISSUE THREAD

**Issue by TonyCr46** [TortugaPower/BookPlayer#470]
I have used Bookplayer for some time and I am quite happy with it. However, I have long been curious (with any audiobook player I have used) if a particular feature could be implemented.

I like to listen to a book before I sleep, and I generally like to set the sleep timer for about 30 minutes. I re-format a lot of my audiobooks using various apps so that the lengths of the chapters are about 30 minutes. This often means merging several original book chapters when the chapter lengths are much shorter. I've often wondered if it would be possible to, in effect, have the app do this. Here's an example:

If the chapters are short and I use the end of chapter option for the sleep timer, I may get only 5-7 minutes on some books. However, if I use one of the specific timer length options (15 minutes, 30 minutes, etc.), it usually stops in the middle of a chapter, a sentence, or even a word. 

It would be nice is if there was an additional option in the sleep timer to select "Proceed to End of Chapter". What would happen is that, if this option is selected, the player would hit the end of the timed period and then continue to the end of the chapter that it is currently playing.

**Comment by pichfl** [comment]
> It would be nice is if there was an additional option in the sleep timer to select "Proceed to End of Chapter". What would happen is that, if this option is selected, the player would hit the end of the timed period and then continue to the end of the chapter that it is currently playing.

So basically this would say "Set time to 30 minutes and if the then playing continues, play until the chapter ends". Very interesting idea!

If someone wants to tackle this: I think this should be an option that is added to the settings screen. The interface for the sleep timer is not ideal right now and this would make it even more complicated.

**Comment by TonyCr46** [comment]
I would say the only drawback to having it in settings is that it is on (or off) all the time. Therefore, if I have a book with a long chapter (or even one that is in just one file), and I want a 30-minute hard stop, I'd have to go back into settings and change it for the one case. I was envisioning it as a checkbox within the timer page. All the current options would remain the same, but there would be an option at the bottom to "Continue to End of Chapter". If that is selected, the player would continue on until the chapter ends after hitting the timer setting. If it is not checked, it stops after the end of the timer setting, just like it does now. The other option might be to allow selecting the current end-of-chapter option _in addition_ to a timer setting.

**Comment by pichfl** [comment]
Good point. But how would that look like? The current interface is not made for additions, only a single item can be selected in the action sheet that opens, which is how action sheets must behave in iOS.
